src/Gestures/GestureController.vala

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2025 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * Authored by: Leonhard Kargl <leo.kargl@proton.me>
+ */
+
+public interface Gala.GestureTarget : Object {
+    public enum UpdateType {
+        START,
+        UPDATE,
+        COMMIT,
+        END
+    }
+
+    /**
+     * The actor manipulated by the gesture. The associated frame clock
+     * will be used for animation timelines.
+     */
+    public abstract Clutter.Actor? actor { get; }
+
+    public virtual void propagate (UpdateType update_type, string id, double progress) { }
+}
```

Comment:
This interface should in a separate file according to out code style rules https://docs.elementary.io/develop/writing-apps/code-style#classes-and-files